AllASPFuncs.py

```diff
@@ -659,16 +659,24 @@ def find_parq(ID, path = '/hpc/group/cosmology/OpenUniverse2024/roman_rubin_cats
     Find the parquet file that contains a given supernova ID.
     '''
     files = os.listdir(path)
+
+    star = True if len(str(ID)) == 11 else False
+    
     if star:
         files = [f for f in files if 'pointsource' in f]
         files = [f for f in files if 'flux' not in f]
     else:
         files = [f for f in files if 'snana' in f]
     files = [f for f in files if '.parquet' in f]
+
     for f in files:
         pqfile = int(f.split('_')[1].split('.')[0])
-        df = open_parq(pqfile, path)
-        if ID in df.id.values:
+        if star:
+            df = pd.read_parquet(path+'/pointsource_'+str(pqfile)+'.parquet', engine='fastparquet')
+        else:
+            df = open_parq(pqfile, path)
```

Comment:
Why do these need different functions?

Ideally, once we've identified the type of the object, the other steps can be general.